framework/src/modules/chain/block_processor_v2.js

```diff
@@ -280,16 +285,19 @@ class BlockProcessorV2 extends BaseBlockProcessor {
 		]);
 
 		this.create.pipe([
-			// Getting the BFT header (maxHeightPreviouslyForged and maxHeightPrevoted)
-			async ({ keypair }) => {
-				const delegatePublicKey = keypair.publicKey.toString('hex');
-				return this.bftModule.computeBFTHeaderProperties(delegatePublicKey);
-			},
 			// Create a block with with basic block and bft properties
-			(data, bftHeader) => this._create({ ...data, ...bftHeader }),
-			async (data, block) => {
-				// Saving maxHeightPreviouslyForged before broadcasting
-				await this.bftModule.saveMaxHeightPreviouslyForged(
+			async data => {
+				const previouslyForged = await this._getPreviouslyForgedMap();
+				const delegatePublicKey = data.keypair.publicKey.toString('hex');
+				const maxHeightPreviouslyForged =
+					previouslyForged[delegatePublicKey] || 0;
```

Comment:
Does this have to default to `0` according to what Jan said?

Reply:
According to Jan, If the delegate is forging for the first time, 0 should be the default value.
However, if the data is corrupted or missing, it has to be highest value possible. Jan is thinking a good default value in that case, but it will be tackled in different issue

Reply:
Alright thanks for the clarification